core/lib/spree/bus.rb

```diff
@@ -9,12 +9,12 @@ module Spree
   # It has some modifications to support internal usage of the legacy event
   # system {see Spree::AppConfiguration#use_legacy_events}.
   Bus = Omnes::Bus.new
-  Bus.define_singleton_method(:publish) do |*args, **kwargs, &block|
+  Bus.define_singleton_method(:publish) do |event, **kwargs|
```

Comment:
👍 changed. Any chance you have some reference material about that?

Reply:
I remember several changes to rails to move away from the dreaded 👻 `class_eval` toward `defined_method` turned down because of that, also from a great talk by Jeremy Evans http://code.jeremyevans.net/presentations/rubykaigi2019/index.html#103

```rb
require 'benchmark/ips'

class C
  define_method :foo do |bar|
    bar
  end
  
  def bar(foo)
    foo
  end
end

c = C.new

Benchmark.ips do |x|
  # Typical mode, runs the block as many times as it can
  x.report("define_method") { c.foo(123) }
  x.report("def") { c.bar(123) }
  x.compare!
end
```

_on ruby 3.1_

```
Warming up --------------------------------------
       define_method     1.342M i/100ms
                 def     1.770M i/100ms
Calculating -------------------------------------
       define_method     13.457M (± 0.6%) i/s -     68.433M in   5.085588s
                 def     17.617M (± 1.0%) i/s -     88.517M in   5.025183s

Comparison:
                 def: 17616583.9 i/s
       define_method: 13456764.2 i/s - 1.31x  (± 0.00) slower
```

Reply:
Many thanks, @elia, interesting stuff!